sympy/polys/polytools.py

```diff
@@ -4427,6 +4426,13 @@ def degree(f, gen=0):
         if gen not in p.gens:
             return S.Zero
 
+    elif len(f.free_symbols) > 1 and  sympify(gen).is_Number and not f.is_Poly:
```

Comment:
This `elif` should read `elif not f.is_Poly and len(f.free_symbols) > 1:` because we already know that `sympify(gen).is_Number` is true because of the `if` clause. Also, there is no need to traverse the expression if we know that f is a Poly (and that, I assume is faster than finding the free symbols) -- put cheaper tests first.

Reply:
Thank you. Will take care of this next time.